docs/ABI/Mangling.rst

```diff
@@ -138,6 +138,10 @@ types where the metadata itself has unknown layout.)
   global ::= type generic-signature 'TH' // key path equality
   global ::= type generic-signature 'Th' // key path hasher
 
+  global ::= protocol 'TL'               // protocol requirements base descriptor
+  global ::= assoc-type-name 'Tl'        // associated type descriptor
```

Comment:
Why not use the same Tq suffix we use for method descriptors? There won't be a conflict and you'll be able to reuse l for other things

Reply:
What about conformances? Are we just going to say you can't add new conformance requirements after the fact, and existing conformances have a canonical ordering?

Reply:
I didn't re-use `Tq` because IIUC the mangling grammar is usually not resolved this way: the suffix always (?) tells you what grammar to expect before it. Now, I am stealing several of the thunk mangling  (for the requirements base, associated type, and possibly associated conformance descriptors), so perhaps I should move out to three-letter discriminators because there aren't that many of these.

Associated conformances are interesting. We need to be able to add new conformances for new defaulted associated types, but we can't add any to existing associated types. Introducing associated conformance descriptors would be the simplest way of ensuring that the addition of new requirements doesn't affect existing clients, so I was planning to investigate that next.

Reply:
If new conformances are only ever added to new associated types, you might be able to get away with not giving them their own descriptors no?

Reply:
If we can find a way to consistently put them at the end, we wouldn't need descriptors for associated conformances. But that gets back to the "ABI history database" issue.